lib.nix

```
@@ -99,14 +101,17 @@ rec
           in attrs // lib.optionalAttrs (lib.hasAttr "package" attrs) {
                         package = removeAttrs attrs.package [ "build" ];
                       };
-        cargotomlss = writeText "foo"
-          (lib.concatStrings (lib.mapAttrsToList
-            (k: v: "${k}\n${builtinz.writeTOML "Cargo.toml-ds-aa" (fixupCargoToml v)}\n")
+
+        # a list of tuples from member to cargo toml:
+        #   "foo-member:/path/to/toml bar:/path/to/other-toml"
+        cargotomlss = lib.concatStringsSep " " (lib.mapAttrsToList
```

Note that [Nix will automatically intersperse with " " when you pass a list of things to a derivation](https://nixos.org/nix/manual/#ssec-derivation) so we could simplify this to:
```suggestion
        cargotomlss = (lib.mapAttrsToList
```